File: ios/ScreenshotExceptions.swift
import ExpoModulesCore

internal final class ViewNotFoundException: GenericException<String?> {
  override var reason: String {
    "View with tag \(param ?? "unknown") not found"
  }
}

internal final class UnsupportedOutputException: GenericException<String?> {
  override var reason: String {
    "Unsupported output format: \(param ?? "unknown")"
  }
}

internal final class TakingScreenshotException: Exception {
  override var reason: String {
    "An error occurred while taking a screenshot. Please rise an issue about it with a reproducible example."
  }
}

File: android/src/main/java/expo/modules/screenshot/ScreenshotExceptions.kt
package expo.modules.screenshot

import expo.modules.kotlin.exception.CodedException

internal class ViewNotFoundException(viewTag: Int) :
    CodedException("View with tag $viewTag not found")
internal class UnsupportedOutputException(format: Output) :
    CodedException("Unsupported output format: $format")
internal class TakingScreenshotException :
    CodedException("An error occurred while taking a screenshot. Please rise an issue about it with a reproducible example.")
